src/main/java/com/faforever/client/preferences/ui/SettingsController.java

```diff
@@ -239,7 +239,14 @@ public void initialize() {
     });
 
     currentThemeChangeListener = (observable, oldValue, newValue) -> themeComboBox.getSelectionModel().select(newValue);
-    selectedThemeChangeListener = (observable, oldValue, newValue) -> uiService.setTheme(newValue);
+    selectedThemeChangeListener = (observable, oldValue, newValue) -> {
+      uiService.setTheme(newValue);
+      if (newValue.isNeedsRestart()) {
+        notificationService.addNotification(new PersistentNotification(i18n.get("theme.needsRestart.message", newValue.getDisplayName()), Severity.WARN,
```

Comment:
when the theme.properties specifies needsRestart=true I get an exception here. I think these are the relevant ones:
`at org.springframework.context.support.AbstractMessageSource.getMessage(AbstractMessageSource.java:153)
	at com.faforever.client.i18n.I18n.get(I18n.java:94)
	at com.faforever.client.i18n.I18n.get(I18n.java:90)
	at com.faforever.client.preferences.ui.SettingsController.lambda$initialize$8(SettingsController.java:245)
[...]
Caused by: java.lang.NumberFormatException: For input string: ""`

Reply:
I propose that we don't leave it up to the theme guy to decide whether or not a restart is neede; he might not even know and it might change in future client releases.

Rather the client should figure out itself whether it's needed or not. In its simplest version: require restart if there's more than just a CSS file.
A future client might be able to reload FXML as well.

Reply:
If he changes fxml files of pop ups or the gif in the ladder1v1 search it might not be needed even tho it is more than just a css. So I say a maintainer should better judge himself. Also concider context menus, changes there also don't require a restart. You are right that in general none CSS changes more often require a restart but this is not 100%of cases the case.

Reply:
No answer means u agree with me

Reply:
Fixed